packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -502,6 +509,34 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
     }
   }
 
+  static _findItemIndexWithKey(
+    props: Props,
+    key: string,
+    hint: ?number,
+  ): ?number {
+    if (hint != null) {
+      const curKey = VirtualizedList._getItemKey(props, hint);
+      if (curKey === key) {
+        return hint;
+      }
+    }
+    for (let ii = 0; ii < props.getItemCount(props.data); ii++) {
+      const curKey = VirtualizedList._getItemKey(props, ii);
+      if (curKey === key) {
+        return ii;
+      }
+    }
+    return null;
+  }
+
+  static _getItemKey(props: Props, index: number): ?string {
+    const item = props.data ? props.getItem(props.data, index) : null;
+    if (item == null) {
+      return null;
```

Comment:
The difference between `_getItemKey` and `_keyExtractor` is that the former will take care of calling props.getItem while the latter doesn't and requires passing the item object (see https://github.com/facebook/react-native/pull/35993/files#diff-b450c5c4ec05ca5221b18a829f764b8397eed35a93f388893debb184591b0207R533). I had a couple places where I needed to find an item key by index so this simplified the code, maybe it could be renamed to make it more explicit like `_getItemKeyForIndex` or `_keyExtractorForIndex`?